metadata/src/main/java/org/apache/kafka/controller/PartitionChangeBuilder.java

```diff
@@ -286,6 +290,8 @@ private ElectionResult electAnyLeader() {
                 .filter(replica -> isAcceptableLeader.test(replica))
                 .findFirst();
             if (uncleanLeader.isPresent()) {
+                // update "UncleanLeaderElectionsPerSec" metric
+                metrics.ifPresent(QuorumControllerMetrics::updateUncleanLeaderElection);
```

Comment:
The metrics for unclean leader elections should be in `ControllerMetadataMetrics`, not here. We can tell if a leader election was unclean if the leader wasn't part of the ISR previously.

We cannot put the metrics here because this will not work on standby controllers (the code only gets executed on the active controller).

Reply:
Good suggestion. Changed to `ControllerMetadataMetrics` in this commit: https://github.com/apache/kafka/pull/16284/commits/12edf2ee2f9ed92f4c4e1177bd1092a2967bf005 . I confirmed now all controllers will show this metric. Thanks.